#--------------------------------------------------------#
#--------------- MAPPING    -----------------------------#
#--------------------------------------------------------#

#--------------- ROOMS -------------------------------#
rooms:
    - name : kitchen
      pose1: [4.70,-3.23] #top_left
      pose2: [4.85,0.53]  #top_right
      pose3: [8.58,0.40] #bottom_right
      pose4: [8.57,-3.12] #bottom_left
    - name : dining_room
      pose1: [4.68,-5.7] #top_left
      pose2: [4.74,-3.18]  #top_right
      pose3: [8.58,-3.32] #bottom_right
      pose4: [8.53,-5.85] #bottom_left
    - name : living_room
      pose1: [4.63,-8.63] #top_left
      pose2: [4.67,-5.80]  #top_right
      pose3: [8.57,-5.80] #bottom_right
      pose4: [8.50,-8.68] #bottom_left
    - name : bedroom
      pose1: [1.38,-8.64] #top_left
      pose2: [1.49,-3.84]  #top_right
      pose3: [4.72,-3.90] #bottom_right
      pose4: [4.60,-8.62] #bottom_left
    - name : bathroom
      pose1: [1.44,-3.74] #top_left
      pose2: [1.55,-0.96]  #top_right
      pose3: [4.79,-1.06] #bottom_right
      pose4: [4.69,-3.80] #bottom_left
    - name : entrance
      pose1: [1.47,-0.97] #top_left
      pose2: [1.52,0.62]  #top_right
      pose3: [3.44,0.55] #bottom_right
      pose4: [3.37,-0.98] #bottom_left
    - name : corridor
      pose1: [3.37,-0.98] #top_left
      pose2: [3.44,0.55]  #top_right
      pose3: [4.85,0.53] #bottom_right
      pose4: [4.78,-1.04] #bottom_left



#--------------- LINKS (DOORS)-----------------------#
connections: #open 0 (False), 1 (True), 2 (Unknown)
    - name: door1
      rooms: ["living_room","bedroom"]
      waypoint1: [5.01,-5.37,1]
      waypoint2: [4.25,-5.33,1]
      door: true
      open: 1
    - name: door2
      rooms: ["bedroom","bathroom"]
      waypoint1: [3.89,-4.33,0.5]
      waypoint2: [3.85,-3.46,1]
      door: true
      open: 1
    - name: door3
      rooms: ["bathroom","kitchen"]
      waypoint1: [4.36,-1.76,0]
      waypoint2: [5.04,-1.73,0]
      door: true
      open: 1
    - name: door4
      rooms: ["entrance","corridor"]
      waypoint1: [2.98,-0.27,0]
      waypoint2: [3.75,-0.27,0]
      door: true
      open: 1




objects:
    - name: bottle
      seen: false
      mapped: true
      pose: [0.0,0.0,0.0]
      room: ''
      furniture: ''
      expected_rooms: ["kitchen","living_room"] # Ordered to most likely to less likely
      expected_furniture: ["kitchen_table","tv_table"]


furniture:
    - name: kitchen_table
      static: true
      mapped: true
      seen: false
      room: kitchen
      pose: [6.4,-2.4]
      orientation: 1.3
      size: [1,0.6,1.0]
    - name: tv_table
      static: true
      mapped: true
      seen: false
      room: living_room
      pose: [5.3,-8.3]
      orientation: -0.2
      size: [0.6,0.6,0.4]




#--------------------------------------------------------#
#--------------- NAVIGATION -----------------------------#
#--------------------------------------------------------#
waypoints:
      - name : door_livingroom_to_bedroom
        pose: [4.17,-6.12]
        room: living_room
      - name : door_bedroom_to_livingroom
        pose: [3.31,-6.04]
        room: bedroom
      - name : door_kitchen_to_bathroom
        pose: [4.77,-2.56]
        room : kitchen
      - name : door_bathroom_to_kitchen
        pose: [3.97,-2.39]
        room : bathroom
      - name : door_bedroom_to_bathroom
        pose: [3.14,-4.81]
        room : bedroom
      - name : door_bathroom_to_bedroom
        pose: [3.32, -3.96]
        room : bathroom


paths: # type 0 door type 1 waypoint
      - name: survey1
        poses:
            - type: 0
              name: door4
              forward: true
            - type: 0
              name: door3
              forward: false
            - type: 0
              name: door2
              forward: false
            - type: 0
              name: door1
              forward: false
            - type: 0
              name: door3
              forward: false



